Docs 3164 secure prop v4.2

Cristian-Venticinque opened the pull request.

Commit:
Added Note clarifying limitations for <key>

Commit:
Added section Arguments reference. Moved note to [NOTE] block.

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
 ----
 
-Note that key sizes might depend on the Java Cryptography Extension (JCE) version. For example, since version 1.8.0_161, the AES key size needs to be of at least 32 characters (256 bytes), instead of 16 for previous versions.
+=== Arguments Reference
```
The first five rows in this table seem confusing. In these rows, what is the relationship between the first and the third columns?

For example, if the argument is `string`, how is it possible that `file` and `file-level`, which are separate arguments, are documented here as Accepted Values for `string`?

The three values are acceptable for decryption mode, but not for the argument `string`. But I don't know how to reword this. I think the accepted values for string, file, and file-level are really the second sentence of the description column entry:

Argument             Description                 Value
string                   Sets...                          An encrypted single-string value provided as argument <value>.

Does that make any sense?

Cristian-Venticinque replied:
I'm reworking the whole to get rid of the third column and clarify difference between arguments and values.

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+|Sets decryption mode to `string`. +
+Encrypts a single string value provided as argument `<value>`.
+|
+* `string`
```
See above.

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+|Sets decryption mode to `file`. +
+Encrypts the value of each property set in the properties file provided as argument `<input file>`.
+|
+* `string`
```
See above.

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+|Sets decryption mode to `file-level`. +
+Encrypts the whole content of the properties file provided as argument `<input file>`.
+|
+* `string`
```
See above.

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+
+Only one of these can be set per operation.
+
+|`encrypt`
```
See above.
I have no idea what we could put in the third column for these. Encrypt and decrypt are two separate arguments with only one value apiece....

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+
+Only one of these can be set per operation.
+
+|`decrypt`
```
ditto

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+Only one of these can be set per operation.
+
+|`<algorithm>`
+|The algorithm to use during the operation.
```
```suggestion
|Specifies the algorithm to use during the operation.
```

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+| See <<supported_modes>> for a full list.
+
+|`<key>`
+|The word used to lock or unlock the file or value during the operation. +
```
```suggestion
|Specifies the word used to lock or unlock the file or value during the operation. +
```

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+|Any string value.
+
+|`<value>`
+|String value that is encrypted or decrypted during the `string` operation.
```
Specifies....
GLOBAL below.
Just to make all Column 2 entries complete, verb-based sentences.

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+
+[NOTE]
+====
+* Key sizes might depend on the Java Cryptography Extension (JCE) version. For example, since version 1.8.0_161, the AES key size needs to be of at least 32 characters (256 bytes), instead of 16 for previous versions.
```
```suggestion
*Beginning with Java Cryptography Extension (JCE) version 1.8.0_161, the AES key size must be of at least 32 characters (256 bytes).
```

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+[NOTE]
+====
+* Key sizes might depend on the Java Cryptography Extension (JCE) version. For example, since version 1.8.0_161, the AES key size needs to be of at least 32 characters (256 bytes), instead of 16 for previous versions.
+* If you use the `$` character in your key, it needs to be escaped with `\`. 
```
```suggestion
* If you use the `$` character in your key, you must precede it with `\`. 
```

kbostrom commented on modules/ROOT/pages/secure-configuration-properties.adoc:
```diff
@@ -349,4 +349,89 @@ The output file is:
+====
+* Key sizes might depend on the Java Cryptography Extension (JCE) version. For example, since version 1.8.0_161, the AES key size needs to be of at least 32 characters (256 bytes), instead of 16 for previous versions.
+* If you use the `$` character in your key, it needs to be escaped with `\`. 
+For example, to use key `myKey#$%123` you need to pass the `<key>` argument as `myKey#\$%123`.
```
```suggestion
For example, to use key `myKey#$%123`, you must specify the `<key>` argument as `myKey#\$%123`.
```

kbostrom reviewed (approved):
Table review

Commit:
Review: Applied editorial feedback.

-Rewrote Secure Properties Tool section.
-Restructured and clarified content.
-Improved clarity of Parameters table.

Commit:
Editorial: title update.

Cristian-Venticinque commented:
Updated branch and solved conflicts in PR #817
